agent/agent.go

```diff
@@ -487,10 +488,21 @@ func (a *Agent) Publisher(ctx context.Context, subscriptionID string) (exec.LogP
 		return nil, nil, err
 	}
 
+	// make little closure for ending the log stream
+	sendCloseMsg := func() {
+		// send a close message, to tell the manager our logs are done
+		publisher.Send(&api.PublishLogsMessage{
```

Comment:
Yeah, you can only use `SendAndClose` for the downstream. The upstream doesn't have that method (it has the cooresponding `RecvAndClose`)